include/pybind11/pytypes.h

```diff
@@ -13,6 +13,7 @@
 #include "buffer_info.h"
 
 #include <exception>
+#include <iostream>
```

Comment:
Can we avoid iostream? This is a rather heavy include and we due a good job making sure use it as little as possible currently